lusidtools/cocoon/cocoon.py

```diff
@@ -901,11 +904,342 @@ async def _construct_batches(
             raise response
 
     # Collects the exceptions as failures and successful calls as values
-    return {
+    returned_response = {
         "errors": [r for r in responses_flattened if isinstance(r, Exception)],
         "success": [r for r in responses_flattened if not isinstance(r, Exception)],
     }
 
+    # For transactions or holdings file types, optionally return unmatched identifiers with the responses
+    if return_unmatched_identifiers is True and file_type in ["transaction", "holding"]:
+        returned_response["unmatched_identifiers"] = unmatched_identifiers(
+            api_factory=api_factory,
+            scope=kwargs.get("scope", None),
+            data_frame=data_frame,
+            mapping_required=mapping_required,
+            file_type=file_type,
+            sync_batches=sync_batches,
+        )
+
+    return returned_response
+
+
+@checkargs
+def unmatched_identifiers(
```

Comment:
Have you considered importing these additional functions from the cocoon/utilities.py, rather than having them defined here? This would:
- make the cocoon.py file more readable
- encourages you to add unittests for some of these

Reply:
I have moved them to utilities but unit tests are needed now

Reply:
tests added